### PR TITLE
feat: MCP server sdk (simple version first)

### PR DESCRIPTION
**Pull request by baxen**

### CODE REVIEW

File: crates/mcp-core/src/protocol.rs [hunk 1]
```diff
@@ -6,42 +6,105 @@ use serde_json::Value;
...
+#[serde(untagged, try_from = "JsonRpcRaw")]
 pub enum JsonRpcMessage {
     Request(JsonRpcRequest),
     Response(JsonRpcResponse),
     Notification(JsonRpcNotification),
     Error(JsonRpcError),
 }
 
+#[derive(Debug, Serialize, Deserialize)]
+struct JsonRpcRaw {
```
Comment (salman1993) [inline crates/mcp-core/src/protocol.rs:53]: where does this get used?

**Review by salman1993 — APPROVED** [review]

File: crates/mcp-server/src/router.rs [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+            response.result =
+                Some(serde_json::to_value(result).map_err(|e| {
+                    RouterError::Internal(format!("JSON serialization error: {}", e))
+                })?);
+
+            Ok(response)
+        }
+    }
+
+    fn handle_resources_list(
```
Comment (salman1993) [inline crates/mcp-server/src/router.rs:183]: getting an error when i test this with MCP inspector now - logs:
```
2024-12-19T15:18:22.942214Z  INFO ThreadId(01) message_processing: crates/mcp-server/src/lib.rs:143: Received request request_id=Some(12) method="resources/list" json={"jsonrpc":"2.0","id":12,"method":"resources/list","params":{}}
2024-12-19T15:18:22.942374Z  INFO ThreadId(01) message_processing: crates/mcp-server/src/lib.rs:173: Sending response response_id=Some(12) json={"jsonrpc":"2.0","id":12,"error":{"code":-32601,"message":"resources/list"}}
2024-12-19T15:18:23.932514Z  INFO ThreadId(01) crates/mcp-server/src/lib.rs:74: incoming message json={"jsonrpc":"2.0","id":13,"method":"resources/list","params":{}}
```

Comment (salman1993) [reply crates/mcp-server/src/router.rs:183]: nvm this works. forgot to pull latest changes 🤦🏼

File: crates/mcp-core/src/protocol.rs (hunk 1 above)
Comment (baxen) [reply crates/mcp-core/src/protocol.rs:53]: this is a deseraialization detail, where we use it to disambiguate when there are so many nulls that you can't tell if its a notification or a request

**Pull request merged**